### PR TITLE
Make IAPIProvider.IsLoggedIn bindable

### PR DESCRIPTION
**Pull request by UselessToucan**
I would like to be able to track auth state using `IsLoggedIn` property without relying on other properties like `LocalUser`.

see https://github.com/UselessToucan/osu/blob/replace_raw_actions_with_commands/osu.Game/Input/Commands/ToggleOnlineOverlayCommand.cs#L20
I know it won't merged, but it is still can be used as an example.

### CODE REVIEW

**Comment by smoogipoo** [comment]
@peppy what do you think of this? I'm sure you already know of the reservations I have w.r.t when do you use `LocalUser.BindValueChanged` vs `OnStageChanged / IOnlineComponent`, and this feels like it might be adding more fuel to that fire.

File: osu.Game/Online/API/DummyAPIAccess.cs [hunk 1]
```diff
@@ -17,7 +17,7 @@ public class DummyAPIAccess : Component, IAPIProvider
             Id = 1001,
         });
 
-        public bool IsLoggedIn => true;
+        public Bindable<bool> IsLoggedIn => new Bindable<bool>(true);
```
Comment (peppy) [inline osu.Game/Online/API/DummyAPIAccess.cs]: incorrectly implemented (see `logout`)

File: osu.Game/Online/API/APIAccess.cs [hunk 2]
```diff
@@ -322,7 +324,9 @@ private bool handleWebException(WebException we)
             return true;
         }
 
-        public bool IsLoggedIn => LocalUser.Value.Id > 1;
+        public Bindable<bool> IsLoggedIn => new Bindable<bool>(getIsLoggedInInternal());
+
+        private bool getIsLoggedInInternal() => LocalUser.Value.Id > 1;
```
Comment (peppy) [inline osu.Game/Online/API/APIAccess.cs]: should not be done like this. follow the same path that `LocalUser` gets set in.

File: osu.Game/Online/API/APIAccess.cs [hunk 3]
```diff
@@ -322,7 +324,9 @@ private bool handleWebException(WebException we)
             return true;
         }
 
-        public bool IsLoggedIn => LocalUser.Value.Id > 1;
+        public Bindable<bool> IsLoggedIn => new Bindable<bool>(getIsLoggedInInternal());
```
Comment (peppy) [inline osu.Game/Online/API/APIAccess.cs]: remove initial value specification here (see next comment).

Comment (peppy) [reply osu.Game/Online/API/APIAccess.cs]: also should be `IBindable`

**Review by peppy — CHANGES_REQUESTED** [review]

**Comment by peppy** [comment]
@smoogipoo i think this is alright, although to me it makes little sense why you'd use this as opposed to `LocalUser` bindable, since basically 100% of the time you want to know the user that is logged in when checking whether logged in.

**Comment by peppy** [comment]
On further discussion we were no longer sure whether this is a good direction. It means a use can either bind to `IsLoggedIn.ValueChanged` or `LocalUser.ValueChanged`. The issue is if a user queries one of these two from the other's `ValueChanged`, what is the expectation? There is going to be one out of two scenarios where the other variable state is stale.

**Pull request closed without merging**